tests/unit/cluster/slot-migration.tcl

```diff
@@ -0,0 +1,363 @@
+proc get_open_slots {srv_idx} {
+    foreach line [split [R $srv_idx CLUSTER NODES] "\n"] {
+        set line [string trim $line]
+        if {$line eq {}} continue
+        if {[regexp {myself} $line] == 0} continue
```

Comment:
I think this part of the function is just `cluster_get_myself` from cluster_util. Ditto about get_cluster_role.